src/data-access/typeORMDataAccess.ts

```diff
@@ -0,0 +1,120 @@
+import { DataSource } from 'typeorm';
+import * as dotenv from 'dotenv';
+import { User } from '../models/typeORMModels';
+import { UserUpdates } from '../types';
+
+dotenv.config();
+
+const username = process.env.DB_USERNAME || '';
+const DBname = process.env.NAME || '';
+const DBpassword = process.env.DB_PASSWORD || '';
+const DBhost = process.env.DB_HOST;
+
+export const AppDataSource = new DataSource({
+  type: 'postgres',
+  host: DBhost,
+  port: 5432,
+  username,
+  password: DBpassword,
+  database: DBname,
+  synchronize: true,
+  logging: true,
+  entities: [User],
+  subscribers: [],
+  migrations: [],
+});
+
+AppDataSource.initialize()
+  .then(() => {
+    console.log('Data Source has been initialized!');
+  })
+  .catch((err) => {
+    console.error('Error during Data Source initialization:', err);
```

Comment:
For future thought: If error occurred during DB initialization, your app hardly could work, couldn't it? You probably need to crash it then.

Reply:
fixed

Reply:
fixed